modules/theAdxBidAdapter.js

```diff
@@ -0,0 +1,524 @@
+                url: content.url,
+                width: content.w,
+                height: content.h
+              };
+            }
+          });
+        }
+
+        let response = {
+          bidderCode: BIDDER_CODE,
+          requestId: request.bidId,
+          cpm: bid.price,
+          width: bidWidth | 0,
+          height: bidHeight | 0,
+          ad: creative,
+          ttl: ttl || 3000,
+          creativeId: bid.crid,
+          netRevenue: true,
+          currency: responseBody.cur,
+          mediaType: mediaType,
+          native: native,
+        };
+        if (mediaType == VIDEO && videoXml) {
+          response.vastUrl = videoXml;
+          response.videoCacheKey = bid.ext.rid;
+        }
+
+        responses.push(response);
+      }
+    }
+    return responses;
+  },
+
+  /**
+   * Register the user sync pixels which should be dropped after the auction.
+   *
+   * @param {SyncOptions} syncOptions Which user syncs are allowed?
+   * @param {ServerResponse[]} serverResponses List of server's responses.
+   * @return {UserSync[]} The user syncs which should be dropped.
+   */
+  getUserSyncs: function (syncOptions, serverResponses) {
+    const syncs = [];
+    if (syncOptions.iframeEnabled) {
+      syncs.push({
+        type: 'iframe',
+        url: '//ssp.theadx.com/async_usersync_iframe.html'
+      });
+    }
+
+    return syncs;
+  },
+
+  /**
+   * @param {TimedOutBid} timeoutData
+   */
+  onTimeout: (timeoutData) => {
+    try {
+      let eventData = {
+        name: 'TIMEOUT',
+        value: timeoutData.length,
+        related_data: timeoutData[0].timeout || config.getConfig('bidderTimeout')
+      };
+      logEvent(eventData, timeoutData);
+    } catch (e) {}
+  },
+
+  /**
+   * @param {TimedOutBid} timeoutData
+   */
+  onBidWon: (bid) => {
+    try {
+      let eventData = {
+        name: 'BID_WON',
+        value: bid.cpm
+      };
+      logEvent(eventData, [bid]);
+    } catch (e) {}
+  },
+
+}
+
+function getLoggingData(event, data) {
+  data = (utils.isArray(data) && data) || [];
+  let params = {};
+  params.project = 'prebid';
+  params.acid = utils.deepAccess(data, '0.auctionId') || '';
+  params.rid = utils.deepAccess(data, '0.requestId') || '';
+  params.ttr = utils.deepAccess(data, '0.timeToRespond') || 0;
+  params.tagId = data.map((adunit) => {
+    return `tag-${utils.deepAccess(adunit, 'params.0.tagId') || 0}:${adunit.adUnitCode}`
+  }).join('|');
+  params.adunit_count = data.length || 0;
+  params.dn = utils.getTopWindowLocation().host || '';
+  params.requrl = utils.getTopWindowUrl() || '';
+  params.event = event.name || '';
+  params.value = event.value || '';
+  if (event.related_data) {
+    params.rd = event.related_data || ''
+  };
+
+  return params;
+}
+
+function logEvent (event, data) {
+  let getParams = {
+    protocol: 'https',
+    hostname: EVENT_PIXEL_URL,
+    search: getLoggingData(event, data)
+  };
+  utils.triggerPixel(url.format(getParams));
```

Comment:
Adapters are prohibited from firing off additional network calls.

Reply:
onTimeout&onBidWon events removed, so its not necessary,removed